# IMAP: Understanding Sample Metadata

Welcome to the "IMAP: Understanding Sample Metadata" repository! This repository is part of the IMAP (Integrated Microbiome Analysis Pipeline) project, aimed at providing essential tools and resources for microbiome data analysis.

## About

Sample metadata plays a crucial role in microbiome studies, providing essential information about sample characteristics, experimental conditions, and other relevant factors. This repository aims to provide comprehensive resources and tutorials on understanding, downloading, exploring, and visualizing sample metadata.

## Key Features

- **Introduction to Metadata**: Learn the fundamentals of metadata and its importance in microbiome studies.
- **Downloading Metadata from SRA BioProject**: Step-by-step guide on how to download sample metadata from the Sequence Read Archive (SRA) BioProject database.
- **Exploring Metadata**: Explore techniques for analyzing and interpreting sample metadata to gain insights into your microbiome dataset.
- **Plotting Sample Collection Points on a Map**: Learn how to plot sample collection points on a map to visualize sample distribution geographically.

## Contents

- **Introduction**: Overview of sample metadata and its significance in microbiome research.
- **Downloading Metadata**: Instructions for downloading sample metadata from SRA BioProject using tools like SRA Toolkit.
- **Exploring Metadata**: Tutorials and code examples for exploring and analyzing sample metadata using Python and R.
- **Plotting Sample Collection Points**: Guides and code examples for plotting sample collection points on a map using libraries like Folium.

## Getting Started

To get started, clone or download this repository to your local machine. Follow the instructions provided in the respective directories to access tutorials, code examples, and resources for understanding sample metadata.

## Dependencies

- Python 3.x
- R (latest version)
- Required Python packages: pandas, matplotlib, seaborn, folium, etc.
- Required R packages: tidyverse, ggplot2, etc.

Ensure that you have the necessary dependencies installed before running any code examples.

## Contributing

Contributions to this repository are welcome! If you have suggestions, improvements, or additional resources related to understanding sample metadata in microbiome studies, feel free to open an issue or submit a pull request.

Happy exploring and analyzing your microbiome data!


<br>

# Related Works
## IMAP GitHub Repos

| IMAP-Repo                                      | Description                                               | GH-Pages                                             |
| :---------------------------------------------- | :--------------------------------------------------------- | :-----------------------------------------------------: |
| [OVERVIEW](https://github.com/tmbuza/imap-project-overview/) | IMAP project overview                                     | [Link](https://tmbuza.github.io/imap-project-overview/) |
| [PART 01](https://github.com/tmbuza/imap-essential-software/) | Software requirement for microbiome data analysis with Snakemake workflows | [Link](https://tmbuza.github.io/imap-essential-software/) |
| [PART 02](https://github.com/tmbuza/imap-sample-metadata/) | Downloading and exploring microbiome sample metadata from SRA Database | [Link](https://tmbuza.github.io/imap-sample-metadata/) |
| [PART 03](https://github.com/tmbuza/imap-download-sra-reads/) | Downloading and filtering microbiome sequencing data from SRA database | [Link](https://tmbuza.github.io/imap-download-sra-reads/) |
| [PART 04](https://github.com/tmbuza/imap-read-quality-control/) | Quality Control of Microbiome Next Generation Sequencing Reads | [Link](https://tmbuza.github.io/imap-read-quality-control/) |
| [PART 05](https://github.com/tmbuza/imap-bioinformatics-mothur/) | Microbial profiling using MOTHUR and Snakemake workflows | [Link](https://tmbuza.github.io/imap-mothur-bioinformatics/) |
| [PART 06](https://github.com/tmbuza/imap-bioinformatics-qiime2/) | Microbial profiling using QIIME2 and Snakemake workflows | [Link](https://tmbuza.github.io/imap-qiime2-bioinformatics/) |
| [PART 07](https://github.com/tmbuza/imap-data-processing/) | Processing Output from 16S-Based microbiome bioinformatics pipelines | [Link](https://tmbuza.github.io/imap-data-preparation/) |
| [PART 08](https://github.com/tmbuza/imap-exploratory-analysis/) | Exploratory Analysis of 16S-Based Microbiome Processed Data | [Link](https://tmbuza.github.io/imap-data-exploration/) |


## Session information

For a detailed overview of the tools and versions suitable for this guide, explore the [session information](session_info.txt).

## Citation
> Please consider citing the [iMAP article](https://rdcu.be/b5iVj) if you find any part of the iMAP practical user guides helpful in your microbiome data analysis.

Buza, T. M., Tonui, T., Stomeo, F., Tiambo, C., Katani, R., Schilling, M., â€¦ Kapur, V. (2019). iMAP: An integrated bioinformatics and visualization pipeline for microbiome data analysis. BMC Bioinformatics, 20. https://doi.org/10.1186/S12859-019-2965-4

## :tada: Raise awareness
> Please help increase awareness of freely available tools for microbiome data analysis.
See [Dimensions of the iMAP article](https://badge.dimensions.ai/details/id/pub.1117740326)
